cpp/oneapi/dal/algo/linear_regression/backend/gpu/train_kernel_norm_eq_dpc.cpp

```diff
@@ -58,7 +58,7 @@ static train_result<Task> call_dal_kernel(const context_gpu& ctx,
     using dal::detail::check_mul_overflow;
 
     using model_t = model<Task>;
-    using model_impl_t = backend::norm_eq_model_impl<Task>;
+    using model_impl_t = detail::model_impl<Task>;
```

Comment:
I do not fully understand why norm_eq_model_impl was changed to model_impl in certain places?
As I understand norm_eq_model_impl specifies that this model was produced using a normal equations method (norm_eq). In the future we might add other methods to solve linear regression problem (QR, gradient descent, ...).
For me, all the *_model_impl classes for various compute methods will have common parts: coefficiens and intercept; but might have other differences.
What I do not understand: is the current version of the code scales well in case of adding other types of *_model_impl?

Reply:
As you have already said there is no many options what can be inside of the model. Only coefficients and possibly intercept values. So there is no way that another implementation of model should be added. In any way it can be restored relatively easy.

Reply:
And btw you can just extend this model with empty tables if needed. Anyway the method of computations is determined based on descriptor, not on runtime info.

Reply:
Discussed offline